File: yukari_bme280.php
<?php

$YUKARI_VOICE_DIR = '/home/tao/yukari_bme280/voices/';
$BME280_PYTHON = '/home/tao/bme280_tao.py';

exec("sudo python ${BME280_PYTHON}", $out);

$temp = explode(',', $out[0])[0];
$hum = explode(',', $out[0])[1];

$voicePath = array("${YUKARI_VOICE_DIR}/now_temp.wav");
$voicePath = getAudioPath($voicePath, $temp);
array_push($voicePath, "${YUKARI_VOICE_DIR}/degrees_hum.wav");
$voicePath = getAudioPath($voicePath, $hum);
array_push($voicePath, "${YUKARI_VOICE_DIR}/percent_end.wav");

$result = implode(' ', $voicePath);

exec("sudo aplay ${result}");

function getAudioPath($array, $num){
    global $YUKARI_VOICE_DIR;
    $integer = explode('.', $num)[0];
    $decimal = explode('.', $num)[1];

    array_push($array,
        "${YUKARI_VOICE_DIR}/num/${integer}.wav",
        "${YUKARI_VOICE_DIR}/dot.wav"
    );

    $decimalArr = str_split($decimal);
    for($i = 0; $i < count($decimalArr); $i++){
        array_push($array, "${YUKARI_VOICE_DIR}/num/{$decimalArr[$i]}.wav");
    }
    return $array;
}

File: yukari_bme280.go
package main

import (
	"fmt"
	"os"
	"os/exec"
	"strings"
)

const YUKARI_VOICE_DIR = "/home/tao/yukari_bme280/voices/"
const BME280_PYTHON = "/home/tao/bme280_tao.py"

func main() {
	out, err := exec.Command("sudo", "python", BME280_PYTHON).Output()
	if err != nil {
		fmt.Errorf("Error: failed get nums from bme280")
		os.Exit(1)
	}
	temp := strings.Split((string)(out), ",")[0]
	hum := strings.Split((string)(out), ",")[1]

	voicePath := []string{"sudo", "aplay", YUKARI_VOICE_DIR + "/now_temp.wav"}
	appendVoicePath(&voicePath, temp)
	voicePath = append(voicePath, YUKARI_VOICE_DIR+"/degrees_hum.wav")
	appendVoicePath(&voicePath, hum)
	voicePath = append(voicePath, YUKARI_VOICE_DIR+"/percent_end.wav")

	exec.Command(voicePath[0], voicePath[1:]...).Run()
}

func appendVoicePath(array *[]string, num string) {
	integer := strings.Split(num, ".")[0]
	decimal := strings.Split(num, ".")[1]

	*array = append(*array,
		YUKARI_VOICE_DIR+"/num/"+integer+".wav",
		YUKARI_VOICE_DIR+"/dot.wav")

	decimalArr := strings.Split(decimal, "")
	for _, v := range decimalArr {
		*array = append(*array, YUKARI_VOICE_DIR+"/num/"+v+".wav")
	}
}
